### PR TITLE
DOC: Fix to lzma.LZMAFile

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/compat/__init__.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def _get_lzma_file(lzma):
 
     Returns
     -------
-    class
+    lzma.LZMAFile
```
Comment (jbrockmendel) [inline pandas/compat/__init__.py]: possibly `type(LZMAFile)`?  usually this would indicate that we're getting an instance of that class

Comment (WillAyd) [reply pandas/compat/__init__.py]: Should annotate the function return here as well

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/compat/__init__.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/compat/__init__.py]: > Should annotate the function return here as well

I thought of that. but the problem is that the function tests for the imports of the ```lzma``` module. If I import it beforehand it kinda losing it's point (I think)

Comment (MomIsBestFriend) [reply pandas/compat/__init__.py]: > Should annotate the function return here as well

@WillAyd I tried it now, I imported the ```lzma``` module at the beginning of the file and annotated ```_get_lzma_file``` with ```-> lzma.LZMAFile```, and ```mypy``` complained:

```
pandas/io/common.py:468: error: "LZMAFile" not callable
``` 

Any thoughts?

**Comment by WillAyd** [comment]
It would need to be Type[lzma.LZMAFile], though not sure if we can do that given this is technically an optional dependency from the stdlib

**Comment by WillAyd** [comment]
Does Type[IO] work? May be the best bet

**Comment by topper-123** [comment]
LZMA has been part of the standard library since python 3.3. We should just remove assume it’s installed, that will simplify this a bit.

**Comment by jbrockmendel** [comment]
> LZMA has been part of the standard library since python 3.3. We should just remove assume it’s installed, that will simplify this a bit.

I'm pretty sure that just recently i had to do an apt-get install to get this working on an ubuntu 18 machine

**Comment by WillAyd** [comment]
We had an issue previously with that assumption, hence the existence of this compat to begin with. See #27575

**Comment by topper-123** [comment]
Ok, that’s surprising. Yeah, changing it isn’t possible then, I guess...

**Comment by MomIsBestFriend** [comment]
> It would need to be Type[lzma.LZMAFile], though not sure if we can do that given this is technically an optional dependency from the stdlib

```mypy```'s log:

```
Success: no issues found in 808 source files
```

> Does Type[IO] work? May be the best bet

```mypy```'s log:

```
pandas/io/common.py:468: error: Too many arguments for "IO"
Found 1 error in 1 file (checked 808 source files)
```

**Comment by MomIsBestFriend** [comment]
> > It would need to be Type[lzma.LZMAFile], though not sure if we can do that given this is technically an optional dependency from the stdlib
> 
> `mypy`'s log:
> 
> ```
> Success: no issues found in 808 source files
> ```
> 
> > Does Type[IO] work? May be the best bet
> 
> `mypy`'s log:
> 
> ```
> pandas/io/common.py:468: error: Too many arguments for "IO"
> Found 1 error in 1 file (checked 808 source files)
> ```

I just found a bug (I think), unless I'm missing something obvious.

When  ``` _get_lzma_file``` is annotated with ```-> Type[IO]```, normally ```mypy``` will say:
```
pandas/io/common.py:468: error: Too many arguments for "IO"
Found 1 error in 1 file (checked 808 source files)
```

But if the ```lzma``` module is imported at the beginning of the file you ```mypy``` will say:
```
Success: no issues found in 808 source files
```

---

So two questions actually:
* Is this a bug?
* What option do we prefer? (if any)

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

Can come back to types later for the whole module if something you are interested in

**Pull request merged**